COM-680 HTS_INDEX - Number of newly diagnosed clients in the facility offered index case testing (ICT)

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-680

tmvumbi2 committed:
COM-680 Add index indicator 1a
Implements the index indicator 1a

tmvumbi2 committed:
COM-680 Add index indicator 1a
Implements the index indicator 1a

lauravignoli commented on metadata/reportssql/index_report_functions.sql:
```diff
@@ -0,0 +1,42 @@
+    patient pat
+WHERE
+    patientGenderIs(pat.patient_id, p_gender) AND
+    patientAgeWhenRegisteredForHivProgramIsBetween(pat.patient_id, p_startAge, p_endAge, p_includeEndAge) AND
```
@tmvumbi2 I checked with Richard and we should check the age at report end date instead of at hiv program registration. Let's chat because I think we've been using this function in reports that are not related to hiv registration date.

lauravignoli commented on metadata/reportssql/index_report_functions.sql:
```diff
@@ -0,0 +1,42 @@
+
+-- getPatientIndexTestingDateOffered
+
+DROP FUNCTION IF EXISTS getPatientIndexTestingDateOffered;
```
@tmvumbi2 Can you please rename to getPatientIndexTestingOfferedDate?

tmvumbi2 committed:
Merge branch 'COM-680' of https://github.com/jembi/cameroon-bahmni-config into COM-680

lauravignoli commented on metadata/reportssql/index_report_functions.sql:
```diff
@@ -0,0 +1,42 @@
+    patientGenderIs(pat.patient_id, p_gender) AND
+    patientAgeWhenRegisteredForHivProgramIsBetween(pat.patient_id, p_startAge, p_endAge, p_includeEndAge) AND
+    getPatientIndexTestingDateOffered(pat.patient_id) IS NOT NULL AND
+    getObsCodedShortNameValue(pat.patient_id, uuidServiceRequired) NOT IN ("Community home", "Community mobile");
```
@tmvumbi2 I chatted to Richard and we shouldn't use "service required" for this. I think a better way would be to check the obs location in getPatientIndexTestingDateOffered? Maybe we could rename to getPatientIndexTestingOfferedDateForNonHomeOrMobileLocation or smthg like that?

tmvumbi2 committed:
COM-680 Use the report end age to calculate patient age
The age disaggregation needs to use the patient's age at the end of the reporting period

tmvumbi2 committed:
COM-680 Use location of the obs "date offered"
To check if the index testing was not offered at a Community Mobile or Community Home

lauravignoli committed:
COM-680 Update Index reports names in template and reports page
